.github/workflows/pipeline-infrastructure-cleardown-non-prod.yaml

```diff
@@ -4,42 +4,55 @@ on:
   delete:
     branches:
       - 'task/*'
+  pull_request:
+    types:
+      - closed
+
 jobs:
   metadata:
-    if: github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]'
+    if: (github.ref_type == 'branch' || github.event.ref_type == 'branch') && (startswith(github.head_ref, 'task/') || startswith(github.event.ref, 'task/')) && github.actor != 'github-merge-queue[bot]'
     name: "Get Metadata"
     uses: NHSDigital/uec-dos-management/.github/workflows/metadata.yaml@latest
+
   destroy:
-    if: github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]'
+    if: (github.ref_type == 'branch' || github.event.ref_type == 'branch') && (startswith(github.head_ref, 'task/') || startswith(github.event.ref, 'task/')) && github.actor != 'github-merge-queue[bot]'
+    name: "Destroy infrastructure on ${{ matrix.environment }} Environment"
+    strategy:
+      matrix:
+        environment: ${{ fromJson(needs.metadata.outputs.environment) }}
     uses: NHSDigital/uec-dos-management/.github/workflows/infrastructure-cleardown.yaml@latest
     needs:
       [
         metadata,
       ]
     with:
-      environment: dev
+      environment: ${{ matrix.environment }}
       workspace: ${{ needs.metadata.outputs.workspace }}
-      domain: "uec-dos-service-management"
+      domain: ${{ needs.metadata.outputs.reponame }}
+      project: ${{ vars.PROJECT }}
       stacks: '["application", "data_migration", "domain-test"]'
     secrets: inherit
 
   cleardown-artefacts:
-    if: github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]'
+    if: needs.metadata.outputs.environment == '["dev"]' && github.event.ref_type == 'branch' && startswith(github.head_ref, 'task/') && github.actor != 'github-merge-queue[bot]'
```

Comment:
`startswith(github.head_ref` needs to be `startswith(github.event.ref` in order to trigger on a branch deletion

Reply:
Thanks